src/java/org/pantsbuild/tools/ivy/DependencyUpdateChecker.java

```diff
@@ -0,0 +1,295 @@
+
+    ResolveReport latestReport = getIvyInstance().resolve(latestModuleDescriptor, resolveOptions);
+
+    displayDependencyUpdates(getResolvedReport(), latestReport);
+    if (showTransitive) {
+      displayNewDependencyOnLatest(getResolvedReport(), latestReport);
+      displayMissingDependencyOnLatest(getResolvedReport(), latestReport);
+    }
+  }
+
+  private void displayDependencyUpdates(ResolveReport originalReport, ResolveReport latestReport) {
+    log("Dependency updates available:");
+    boolean dependencyUpdateDetected = false;
+    for (Object ivyNode : latestReport.getDependencies()) {
+      IvyNode latest = (IvyNode) ivyNode;
+      for (Object ivyNode2 : originalReport.getDependencies()) {
+        IvyNode originalDependency = (IvyNode) ivyNode2;
+        if (originalDependency.getModuleId().equals(latest.getModuleId())) {
+          if (!originalDependency.getResolvedId().getRevision()
+              .equals(latest.getResolvedId().getRevision())) {
+            // is this dependency a transitive dependency or a direct dependency
+            // (unfortunatly the .isTranstive() method doesn't have the same meaning)
+            boolean isTransitiveDependency =
+                latest.getDependencyDescriptor(latest.getRoot()) == null;
+            if ((!isTransitiveDependency) || (isTransitiveDependency && showTransitive)) {
+              StringBuilder sb = new StringBuilder();
+              sb.append(INDENTATION)
+                  .append(originalDependency.getResolvedId().getOrganisation())
+                  .append('#')
+                  .append(originalDependency.getResolvedId().getName())
+                  .append(isTransitiveDependency ? " (transitive)" : "")
+                  .append(INDENTATION)
+                  .append(originalDependency.getResolvedId().getRevision())
+                  .append(" -> ")
+                  .append(latest.getResolvedId().getRevision());
+              log(sb.toString());
+              dependencyUpdateDetected = true;
+            }
+          }
+        }
+      }
+    }
+    if (!dependencyUpdateDetected) {
+      log(INDENTATION + "All dependencies are up to date");
+    }
+  }
+
+  private void displayMissingDependencyOnLatest(ResolveReport originalReport,
+      ResolveReport latestReport) {
+    List<ModuleRevisionId> listOfMissingDependencyOnLatest = new ArrayList<>();
+    for (Object ivyNode : originalReport.getDependencies()) {
+      IvyNode originalDependency = (IvyNode) ivyNode;
+      boolean dependencyFound = false;
+      for (Object ivyNode2 : latestReport.getDependencies()) {
+        IvyNode latest = (IvyNode) ivyNode2;
+        if (originalDependency.getModuleId().equals(latest.getModuleId())) {
+          dependencyFound = true;
+        }
+      }
+      if (!dependencyFound) {
+        listOfMissingDependencyOnLatest.add(originalDependency.getId());
+      }
+    }
+
+    if (listOfMissingDependencyOnLatest.size() > 0) {
```

Comment:
Refactoring this such that the return code of the binary can be meaningful (ie, returning non-zero if updates are available) might be interesting.